### PR TITLE
Changes to avoid self fund on same addres at grants

### LINKED ISSUE #5459: Grant owners cannot self-fund grants via recurring subscription
**Describe the bug**
If a grant owner attempts to self fund a grant, the subscriptions will fail in the subminer due to a check in the grant contract code that the origin and destination addresses are not the same: https://github.com/danlipert/grants1337/blob/specify-relayer/contracts/Subscription.sol#L186

**To Reproduce**
Create a grant and then with the same recipient address create a recurring subscription. Run the subminer and the metatx will fail.

**Expected behavior**
Given that this code is already in the deployed contracts, changing it now may not be wise. I suggest we popup a message or disable recurring subscriptions in this case or something similar instead of changing the contract. @owocki what are your thoughts?

### PR DESCRIPTION
**Pull request by zoek1**
##### Description
When a grant owner try to fund a grant with the recurring option selected, one message shol=uld prompt the user that switch the account.

Fund Grant _ Gitcoin

Also at the backen side there one rule to redirect to grant detail page when the grant admin address and the contributor address are the same.

##### Refers/Fixes
#5459

### CODE REVIEW

File: app/assets/v2/js/grants/fund.js [hunk 1]
```diff
@@ -199,6 +199,10 @@ $(document).ready(function() {
         let approvalSTR = realApproval.toLocaleString('fullwide', { useGrouping: false });
 
         web3.eth.getAccounts(function(err, accounts) {
+          if (data.recurring_or_not == 'recurring' && grant_admin_addres === accounts[0]){
+            _alert({ message: gettext('Grant owners cannot self-fund grants. Please change your account.') }, 'error');
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/fund.js]: update it to state grant owners cannot use recurring option while self funding

File: app/grants/templates/grants/fund.html [hunk 2]
```diff
@@ -354,6 +354,8 @@ <h5 class="form__label">{% trans "Gas Settings" %}</h5>
     {% include 'grants/shared/shared_scripts.html' %}
 
     <script>
+      const grant_admin_addres = '{{ grant.admin_address }}';
```
Comment (thelostone-mc) [inline app/grants/templates/grants/fund.html]: `grant_admin_addres` -> `grant_admin_address`

File: app/assets/v2/js/grants/fund.js (hunk 1 above)
Comment (thelostone-mc) [reply app/assets/v2/js/grants/fund.js]: `grant_admin_addres` -> `grant_admin_address`

**Comment by zoek1** [comment]
> would it be possible to set the default to One Time Payment in the UI
if the grant owner logs in with the same wallet address as the grant admin address?

Sure, should the page requests the account address and update to one time payment as soon the page loads or should wait until the user click the button, display the alert and then change the option to one time payment?

@thelostone-mc

**Comment by zoek1** [comment]
> would it be possible to set the default to One Time Payment in the UI
if the grant owner logs in with the same wallet address as the grant admin address?

@thelostone-mc i added this behavior to the code, when is detected a self fund change inmediately to 'once opton'.

File: app/grants/views.py [hunk 3]
```diff
@@ -518,7 +517,7 @@ def grant_fund(request, grant_id, grant_slug):
             'grant': grant,
             'text': _('You already have an active subscription for this grant.')
         }
-        return TemplateResponse(request, 'grants/shared/error.html', params)
+        # return TemplateResponse(request, 'grants/shared/error.html', params)
```
Comment (thelostone-mc) [inline app/grants/views.py]: Why was this commented out ?

**Comment by thelostone-mc** [comment]
@zoek1 it's almost wrapped up ! just had one minor comment which I've 
Otherwise LGTM to me

File: app/grants/views.py (hunk 3 above)
Comment (zoek1) [reply app/grants/views.py]: :scream: Sorry, I commented this to reuse some grants, but I comment it again!

File: app/assets/v2/js/grants/fund.js [hunk 4]
```diff
@@ -199,6 +200,11 @@ $(document).ready(function() {
         let approvalSTR = realApproval.toLocaleString('fullwide', { useGrouping: false });
 
         web3.eth.getAccounts(function(err, accounts) {
+          if (data.recurring_or_not == 'recurring' && grant_admin_address === accounts[0]) {
```
Comment (octavioamu) [inline app/assets/v2/js/grants/fund.js]: shouldn't we check the user instead? I believe the address for the owner can be set to any address. Or also another grant team member can still go and auto fund.

Comment (danlipert) [reply app/assets/v2/js/grants/fund.js]: @octavioamu self funding with a different address is okay - theres actually a limitation in the contract itself that causes an error when you self fund with the grant admin's address

Comment (zoek1) [reply app/assets/v2/js/grants/fund.js]: @octavioamu @danlipert is it okay the current validation?

**Comment by thelostone-mc** [comment]
Hey @zoek1 so we had to block grant owners from funding their own grant as explained here
https://github.com/gitcoinco/web/pull/5807
Also we had to hotpatch a fix in to ensure that  one time payment loads first by default as users preferred using that

**Comment by zoek1** [comment]
Definetly improves the flow, happy for that!  🙂

**Pull request closed without merging**